File: phase_2_web_App/frontend/src/types/task.ts
export interface Task {
  id: string;
  title: string;
  description?: string;
  status: string;  // "pending" or "completed"
  priority?: string;
  due_date?: string;
  created_at: string;
  updated_at: string;
  user_id: string;
}

// Helper to check if task is completed
export const isTaskCompleted = (task: Task): boolean => {
  return task.status === 'completed';
};

File: phase_2_web_App/backend/app/models/task.py
from sqlmodel import Field, SQLModel
from uuid import UUID, uuid4
from datetime import datetime, timezone

def utc_now():
    return datetime.now(timezone.utc)


class Task(SQLModel, table=True):
    id: UUID = Field(default_factory=uuid4, primary_key=True)
    user_id: UUID = Field(foreign_key="user.id")
    title: str
    description: str | None = None
    status: str = "pending"
    priority: str | None = None
    due_date: datetime | None = None
    created_at: datetime = Field(default_factory=utc_now)
    updated_at: datetime = Field(default_factory=utc_now)
